### PR TITLE
Allow only one prototool.yaml file

### PR DESCRIPTION
**Pull request by peter-edge**
Per #10 and #93, this change does the bare minimum to enforce only one `prototool.yaml` file per command. There is further refactoring that can be done after this, but this propagates the changes through the codebase while retaining the multiple `prototool.yaml` file support in the implementation for now. Putting this up for discussion, it also helps simplify the logic for #131.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,35 +88,7 @@ The command `prototool init` will generate a config file in the current director
...
-│   └── prototool.yaml
-├── c
-│   ├── file.proto
-│   └── i
-│       └── file.proto
-└── prototool.yaml
-```
-
-Everything under `a/d` will use `a/d/prototool.yaml`, everything under `b`, `b/g/h` will use `b/prototool.yaml`, and everything under `a`, `a/e`, `a/f`, `c`, `c/i` will use `prototool.yaml`. See [internal/file/testdata](internal/file/testdata) for the most current example.
+If multiple `prototool.yaml` files are found that match the input directory or files, an error will be returned. We have an ongoing discussion about whether to allow multiple `prototool.yaml` files, see [this issue](https://github.com/uber/prototool/issues/10) for more details.
```
Comment (amckinney) [inline README.md]: What does it mean for `prototool.yaml` files to match the input directory or files? 

Seeing as you cannot have more than one `prototool.yaml` file in a given directory, could we simply just use the `prototool.yaml` file that corresponds to where the command was run, and ignore all others? For example, in your removed tree example, running `prototool <cmd>` in `b/g` should just use the `prototool.yaml` found there (and not the one included in the root). Similarly, running the same command at the root would use the root `prototool.yaml` and ignore all others.

Let me know if this isn't what you meant - just trying to determine when an error would occur and whether or not it's necessary.

File: internal/protoc/compiler.go [hunk 2]
```diff
@@ -168,31 +164,27 @@ func (c *compiler) Compile(protoSets ...*file.ProtoSet) (*CompileResult, error)
...
-			return nil, err
-		}
-		for _, cmdMeta := range cmdMetas {
-			cmdMetaStrings = append(cmdMetaStrings, cmdMeta.String())
-		}
-		cleanCmdMetas(cmdMetas)
+func (c *compiler) ProtocCommands(protoSet *file.ProtoSet) ([]string, error) {
+	// we end up calling the logic that creates temporary files for file descriptor sets
+	// anyways, so we need to clean them up with cleanCmdMetas
+	// this logic could be simplified to have a "dry run" option, but ProtocCommands
```
Comment (amckinney) [inline internal/protoc/compiler.go]: nit: Now that this uses the `dry-run` flag, could we remove this comment? My fault for not handling in https://github.com/uber/prototool/pull/135

Comment (peter-edge) [reply internal/protoc/compiler.go]: Can't remove this yet, needs some refactoring, this still creates temporary files.

File: internal/file/proto_set_provider.go [hunk 3]
```diff
@@ -52,7 +52,45 @@ func newProtoSetProvider(options ...ProtoSetProviderOption) *protoSetProvider {
...
+	case 0:
+		return nil, fmt.Errorf("no proto files found for dirPath %q", dirPath)
+	case 1:
+		return protoSets[0], nil
+	default:
+		configDirPaths := make([]string, 0, len(protoSets))
+		for _, protoSet := range protoSets {
+			configDirPaths = append(configDirPaths, protoSet.Config.DirPath)
+		}
+		return nil, fmt.Errorf("multiple configuration files found in directories %v for proto files found for dirPath %q but expected exactly one configuration file", configDirPaths, dirPath)
```
[the remark this answers is not in the document]
Comment (peter-edge) [reply internal/file/proto_set_provider.go]: Done

File: internal/file/proto_set_provider.go [hunk 4]
```diff
@@ -52,7 +52,45 @@ func newProtoSetProvider(options ...ProtoSetProviderOption) *protoSetProvider {
...
+	case 0:
+		return nil, fmt.Errorf("no proto files found for filePaths %v", filePaths)
+	case 1:
+		return protoSets[0], nil
+	default:
+		configDirPaths := make([]string, 0, len(protoSets))
+		for _, protoSet := range protoSets {
+			configDirPaths = append(configDirPaths, protoSet.Config.DirPath)
+		}
+		return nil, fmt.Errorf("multiple configuration files found in directories %v for proto files found for filePaths %v but expected exactly one configuration file", configDirPaths, filePaths)
```
[the remark this answers is not in the document]
Comment (peter-edge) [reply internal/file/proto_set_provider.go]: Done

**Pull request merged**